src/components/IdleTimeoutModal.js

```diff
@@ -0,0 +1,105 @@
+import _ from 'lodash/fp'
+import { useEffect, useState } from 'react'
+import { div, h, iframe } from 'react-hyperscript-helpers'
+import ButtonBar from 'src/components/ButtonBar'
+import Modal from 'src/components/Modal'
+import { getUser } from 'src/libs/auth'
+import colors from 'src/libs/colors'
+import { authStore, lastActiveTimeStore } from 'src/libs/state'
+import * as Utils from 'src/libs/utils'
+
+
+const displayRemainingTime = remainingSeconds => {
+  return _.join(':', [
+    `${Math.floor(remainingSeconds / 60).toString().padStart(2, '0')}`,
+    `${Math.ceil(remainingSeconds % 60).toString().padStart(2, '0')}`
+  ])
+}
+
+const setLastActive = lastActive => lastActiveTimeStore.update(existing => ({ ...existing, [getUser().id]: lastActive }))
+
+const IdleTimeoutModal = ({
+  timeout = Utils.hhmmssToMs({ mm: 15 }),
+  countdownStart = Utils.hhmmssToMs({ mm: 3 }), emailDomain = ''
+}) => {
+  const { isSignedIn, profile: { email }, user: { id } } = Utils.useStore(authStore)
+  const isClinicalDomain = email && emailDomain ? email.includes(`@${emailDomain}`) : false
```

Comment:
nit: Since this is computing a boolean, it could just use `&&` instead of a ternary.